docs/generate-time-series/javascript.rst

```diff
@@ -0,0 +1,241 @@
+.. _gen-ts-javascript:
+
+==========================================
+Generate time series data using Javascript
+==========================================
+
+This tutorial will show you how to generate some :ref:`experimental time series
+data <gen-ts>` by consuming telemetry data from the `International Space Station`_
+using Javascript, specifically `Node.js`_.
+
+.. SEEALSO::
+
+    :ref:`gen-ts`
+
+.. rubric:: Table of contents
+
+.. contents::
+   :local:
+
+
+Prerequisites
+=============
+
+CrateDB must be :ref:`installed and running <install-run>`.
+
+Make sure you're running up-to-date `Node`_.
+
+Also make sure you're running up-to-date `npm`_. To upgrade npm:
+
+.. code-block:: console
+
+    $ npm install -g npm@latest
+
+Use npm to install the `node-postgres`_ library:
+
+.. code-block:: console
+
+    $ npm install pg
+
+You will also be using the `Axios`_ library to make HTTP queries. Install the ``axios``
+library with npm:
+
+.. code-block:: console
+
+    $ npm install axios
+
+
+Both the ``axios`` and ``node-postgres`` libraries operate using `Promises`_. Promises are
+a way of encapsulating the eventual result of an asynchronous operation. Both ``axios``, when
+retrieving content from a URL, and ``node-postgres``, when connecting and querying a database,
+use Promises as a way of asynchronously doing those operations.
```

Comment:
would delete the bit starting "Both"

if you want to include info about what they use promises for you could reword the first sentence more like this:

"The ``node-postgres`` and ``axios` libraries both use `Promises`_ when performing network operations."